ethcore/src/engines/clique/block_state.rs

```diff
@@ -0,0 +1,207 @@
+// Copyright 2015-2019 Parity Technologies (UK) Ltd.
+// This file is part of Parity Ethereum.
+
+// Parity Ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::{HashMap, VecDeque};
+use std::time::{Duration, SystemTime};
+use std::time::UNIX_EPOCH;
+
+use ethereum_types::Address;
+use rand::Rng;
+
+use engines::clique::{VoteType, DIFF_INTURN, DIFF_NOTURN, NONCE_AUTH_VOTE, NONCE_DROP_VOTE, NULL_AUTHOR, SIGNING_DELAY_NOTURN_MS};
+use engines::clique::util::{extract_signers, recover_creator};
+use error::Error;
+use types::header::Header;
+
+#[derive(Clone, Debug)]
+pub struct CliqueBlockState {
```

Comment:
Please remember to add docs for all public fields and functions before merging!

Reply:
Yeah, that is good point. However shouldn't https://github.com/paritytech/parity-ethereum/blob/master/ethcore/src/lib.rs#L17 emit warnings for these? I can't find these at least